graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/functions/messages/RemoveFieldsByValue.java

```diff
@@ -0,0 +1,89 @@
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+import static org.graylog.plugins.pipelineprocessor.ast.functions.ParameterDescriptor.type;
+
+public class RemoveFieldsByValue extends AbstractFunction<Void> {
+    public static final String NAME = "remove_fields_by_value";
+    private static final String VALUES_ARG = "values";
+    private static final String PATTERN_ARG = "pattern";
+    private final ParameterDescriptor<String, Pattern> regexParam;
+    private final ParameterDescriptor<List, List> valuesParam;
+    private final ParameterDescriptor<Message, Message> messageParam;
+
+    public RemoveFieldsByValue() {
+        regexParam = ParameterDescriptor.string(PATTERN_ARG, Pattern.class)
+                .optional()
+                .transform(Pattern::compile)
+                .description("A regex specifying field values to be removed.").build();
+        valuesParam = type(VALUES_ARG, List.class).optional().description("A list of exact values of fields to be removed.").build();
+        messageParam = type("message", Message.class).optional().description("The message to use, defaults to '$message'").build();
+    }
+
+    @Override
+    public Void evaluate(FunctionArgs args, EvaluationContext context) {
+        final Message message = messageParam.optional(args, context).orElse(context.currentMessage());
+        if (regexParam.optional(args, context).isPresent()) {
+            removeRegex(message, regexParam.optional(args, context).get());
+        }
+        if (valuesParam.optional(args, context).isPresent()) {
+            removeNames(message, valuesParam.optional(args, context).get());
+        }
+        return null;
+    }
+
+    private void removeRegex(Message message, Pattern pattern) {
+        message.getFields().entrySet().stream()
+                .filter(entry -> entry.getValue() instanceof String && pattern.matcher((String) entry.getValue()).matches())
+                .toList() // required to avoid ConcurrentModificationException
+                .forEach(entry -> message.removeField(entry.getKey()));
+    }
+
+    private void removeNames(Message message, List values) {
+        message.getFields().entrySet().stream()
+                .filter(entry -> entry.getValue() instanceof String && values.contains(entry.getValue()))
+                .toList() // required to avoid ConcurrentModificationException
+                .forEach(entry -> message.removeField(entry.getKey()));
+    }
+
+    @Override
+    public FunctionDescriptor<Void> descriptor() {
+        return FunctionDescriptor.<Void>builder()
+                .name(NAME)
+                .returnType(Void.class)
+                .params(ImmutableList.of(regexParam, valuesParam, messageParam))
+                .description("Removes fields whose value match a value in the values list or the regex pattern, unless the field name is reserved. If no specific message is provided, it uses the currently processed message.")
```

Comment:
Wording nit: It might be a slight improvement to avoid one of the usages of `value` in the first sentence.
```suggestion
                .description("Removes fields whose value exists in the supplied list or the regex pattern, unless the field name is reserved. If no specific message is provided, it uses the currently processed message.")
```